File: ALifeCommon/src/main/java/com/cristiano/alife/instructions/transfers/MovByteInstruction.java
package com.cristiano.alife.instructions.transfers;

import com.cristiano.alife.consts.ALifeConsts;
import com.cristiano.alife.world.BaseInstruction;
import com.cristiano.alife.world.IOrganismo;
import com.cristiano.alife.world.IWorld;
//to <- byte
public class MovByteInstruction extends BaseInstruction  {
	public MovByteInstruction(IWorld mundo) {
		super(mundo);
	}
	@Override
	public void executa(IOrganismo o) {
		int inst=getByteOrganismo(o,1);
		inst=mundo.getMutation().mutateInstruction(inst,o);
		o.setReg(getByteOrganismo(o,2), inst);
	}
	
	@Override
	public int getStep() {
		return 3;
	}
	
	@Override
	public String getDescription(IOrganismo o,int ip){
		return comment(name+" "+o.getMemory(ip+1)+","+ALifeConsts.getLetter(o.getMemory(ip+2)),ALifeConsts.getLetter(o.getMemory(ip+2))+" <- "+o.getMemory(ip+1)+"");
	}
}

File: netAvida/netAvida.Backend/instructions/impl/transfers/MovByteInstruction.cs
﻿using netAvida.backend;
using netAvida.backend.interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace netAvida.Backend.instructions.impl
{
    public class MovByteInstruction : BaseInstruction
    {
        public MovByteInstruction(IWorld mundo): base(mundo)
        {
        }

    public override void executa(IOrganismo o)
        {
            int inst = getByteOrganismo(o, 1);
            inst = mundo.getMutation().mutateInstruction(inst, o);
            o.setReg(getByteOrganismo(o, 2), inst);
        }

    public override int getStep()
        {
            return 3;
        }

    public override String getDescription(IOrganismo o, int ip)
        {
            return comment(name + " " + o.getMemory(ip + 1) + "," + ALifeConsts.getLetter(o.getMemory(ip + 2)), ALifeConsts.getLetter(o.getMemory(ip + 2)) + " <- " + o.getMemory(ip + 1) + "");
        }
    }
}
